### DISCUSSION TITLE
defaultLocale in the bundle

### DISCUSSION THREAD

**Discussion by diogen737** [js-fns/date-fns#3262] [Q&A]
First off - thanks for a great lib!

I need some clarification here.
Let's say I know what locale I'll be using throughout the app at the build time, so I do this:
```typescript
import { setDefaultOptions } from 'date-fns';
import { de } from 'date-fns/locale';

setDefaultOptions({ locale: de });
```

Yet the `en-US` end up in the bundle anyway.

![Screenshot 2022-10-13 at 18 07 45](https://user-images.githubusercontent.com/25563221/195648883-0f6da260-a5fb-4de8-b57a-c79043e0bd0d.png)

Is there a way to tell date-fns/bundler to truly set one default locale once and for all?

**Comment by fturmel** [comment]
`en-US` is the only hard-coded locale in the library itself, so it's always going to be included by default. It shouldn't matter much for most people since it's only about 5.5 Kb gzipped.

https://github.com/date-fns/date-fns/blob/fadbd4eb7920bf932c25f734f3949027b2fe4887/src/_lib/defaultLocale/index.ts#L1

There is official [documentation](https://date-fns.org/docs/webpack) to exclude locales with Webpack's ContextReplacementPlugin. I don't think the approach will allow you to strip out `en-US` though.

There might be other (better?) ways, but I did successfully manage to use the [string-replace-loader](https://github.com/Va1/string-replace-loader) Webpack loader to change the default locale to `de` with the following rule:

```js
{
  test: /_lib\/defaultLocale\/index\.js$/,
  loader: 'string-replace-loader',
    options: {
      search: '/locale/en-US/',
      replace: '/locale/de/',
    },
}
```

<img width="572" alt="Screenshot 2022-10-30 at 12 10 22 PM" src="https://user-images.githubusercontent.com/4009209/198889127-88eeb4e5-7fc4-489c-b9c8-e43f9aa16394.png">

You won't need to explicitly set `de` as the default locale anymore either:

```js
import { format } from 'date-fns';

format(new Date(), 'PPPP') // Sonntag, 30. Oktober 2022
```
